## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 448077`
Commit: [`4a0c1b4d5eb65ecde4a360dfb047815561c71dcc`](https://github.com/NixOS/nixpkgs/commit/4a0c1b4d5eb65ecde4a360dfb047815561c71dcc) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/4a0c1b4d5eb65ecde4a360dfb047815561c71dcc..pull/448077/head))
Merge: [`88925e0ec9295171a6b352f3fbbc06f62950d7e7`](https://github.com/NixOS/nixpkgs/commit/88925e0ec9295171a6b352f3fbbc06f62950d7e7)

Logs: https://github.com/Sigmanificient/nixpkgs-review-gha/actions/runs/18250730662


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>rpcs3</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rpcs3</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>rpcs3</summary>
<pre>  388 | static ASMJIT_INLINE_NODEBUG constexpr Mem ptr(uint64_t base) noexcept { return Mem(base); }
      |                                                ~~~~~~~~~^~~~
[ 13%] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/__/__/Utilities/Thread.cpp.o
[ 13%] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/__/__/Utilities/version.cpp.o
[ 13%] Building CXX object rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/__/Crypto/aes.cpp.o
In file included from /build/source/Utilities/JITLLVM.cpp:50:
/build/source/rpcs3/Emu/../Emu/CPU/Backends/AArch64/AArch64Common.h: At global scope:
/build/source/rpcs3/Emu/../Emu/CPU/Backends/AArch64/AArch64Common.h:35:24: warning: &#x27;aarch64::spr_asm_names&#x27; defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable8;;]
   35 |     static const char* spr_asm_names[] =
      |                        ^~~~~~~~~~~~~
/build/source/rpcs3/Emu/../Emu/CPU/Backends/AArch64/AArch64Common.h:30:24: warning: &#x27;aarch64::spr_names&#x27; defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable8;;]
   30 |     static const char* spr_names[] =
      |                        ^~~~~~~~~
/build/source/rpcs3/Emu/../Emu/CPU/Backends/AArch64/AArch64Common.h:23:24: warning: &#x27;aarch64::gpr_names&#x27; defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-variable-Wunused-variable8;;]
   23 |     static const char* gpr_names[] =
      |                        ^~~~~~~~~
make[2]: *** [rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/build.make:625: rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/__/__/Utilities/JITLLVM.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1125: rpcs3/Emu/CMakeFiles/rpcs3_emu.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/by-name/rp/rpcs3/package.nix
+++ b/pkgs/by-name/rp/rpcs3/package.nix
@@ -2,7 +2,6 @@
   lib,
   stdenv,
   fetchFromGitHub,
-  fetchpatch2,
   nix-update-script,
   cmake,
   pkg-config,
@@ -46,23 +45,15 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "rpcs3";
-  version = "0.0.37";
+  version = "0.0.38";
 
   src = fetchFromGitHub {
     owner = "RPCS3";
     repo = "rpcs3";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-/ve1qe76Rc+mXHemq8DI2U9IP6+tPV5m5SNh/wmppEw=";
+    hash = "sha256-HaguOzCN0/FvAb0b4RZWnw9yvVum14wEj26WnqOnSag=";
     fetchSubmodules = true;
   };
-
-  patches = [
-    (fetchpatch2 {
-      # https://github.com/RPCS3/rpcs3/pull/17316
-      url = "https://github.com/RPCS3/rpcs3/commit/bad6e992586264344ee1a3943423863d2bd39b45.patch?full_index=1";
-      hash = "sha256-rSyA1jcmRiV6m8rPKqTnDFuBh9WYFTGmyTSU2qrd+Go=";
-    })
-  ];
 
   passthru.updateScript = nix-update-script { };
 

--- a/pkgs/by-name/rp/rpcs3/package.nix
+++ b/pkgs/by-name/rp/rpcs3/package.nix
@@ -33,6 +33,13 @@
   waylandSupport ? true,
   wayland,
   wrapGAppsHook3,
+  miniupnpc,
+  rtmidi,
+  asmjit,
+  glslang,
+  zstd,
+  hidapi,
+  vulkan-memory-allocator,
 }:
 
 let
@@ -80,6 +87,12 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeBool "USE_SYSTEM_SDL" true)
     (lib.cmakeBool "USE_SYSTEM_OPENCV" true)
     (lib.cmakeBool "USE_SYSTEM_CUBEB" true)
+    (lib.cmakeBool "USE_SYSTEM_MINIUPNPC" true)
+    (lib.cmakeBool "USE_SYSTEM_RTMIDI" true)
+    (lib.cmakeBool "USE_SYSTEM_GLSLANG" true)
+    (lib.cmakeBool "USE_SYSTEM_ZSTD" true)
+    (lib.cmakeBool "USE_SYSTEM_HIDAPI" true)
+    (lib.cmakeBool "USE_SYSTEM_VULKAN_MEMORY_ALLOCATOR" true)
     (lib.cmakeBool "USE_SDL" true)
     (lib.cmakeBool "WITH_LLVM" true)
     (lib.cmakeBool "BUILD_LLVM" false)
@@ -121,6 +134,13 @@ stdenv.mkDerivation (finalAttrs: {
     libSM
     opencv.cxxdev
     cubeb
+    miniupnpc
+    rtmidi
+    asmjit
+    glslang
+    zstd
+    hidapi
+    vulkan-memory-allocator
   ]
   ++ lib.optional faudioSupport faudio
   ++ lib.optionals waylandSupport [

